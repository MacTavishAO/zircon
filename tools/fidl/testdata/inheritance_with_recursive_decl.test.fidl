library fidl.test.json;

protocol Parent {
    First(request<Parent> request);
};

protocol Child {
    compose Parent;
    Second(request<Parent> request);
};
