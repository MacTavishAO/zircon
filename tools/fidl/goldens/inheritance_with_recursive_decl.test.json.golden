{
  "version": "0.0.1",
  "name": "fidl.test.json",
  "library_dependencies": [],
  "bits_declarations": [],
  "const_declarations": [],
  "enum_declarations": [],
  "experimental_resource_declarations": [],
  "interface_declarations": [
    {
      "name": "fidl.test.json/Parent",
      "location": {
        "filename": "inheritance_with_recursive_decl.test.fidl",
        "line": 3,
        "column": 10,
        "length": 6
      },
      "methods": [
        {
          "ordinal": 2566928499103139852,
          "name": "First",
          "location": {
            "filename": "inheritance_with_recursive_decl.test.fidl",
            "line": 4,
            "column": 5,
            "length": 5
          },
          "has_request": true,
          "maybe_request": [
            {
              "type": {
                "kind": "request",
                "subtype": "fidl.test.json/Parent",
                "nullable": false
              },
              "name": "request",
              "location": {
                "filename": "inheritance_with_recursive_decl.test.fidl",
                "line": 4,
                "column": 27,
                "length": 7
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 4
              }
            }
          ],
          "maybe_request_payload": "fidl.test.json/SomeLongAnonymousPrefix0",
          "maybe_request_type_shape_v1": {
            "inline_size": 24,
            "alignment": 8,
            "depth": 0,
            "max_handles": 1,
            "max_out_of_line": 0,
            "has_padding": true,
            "has_flexible_envelope": false,
            "is_resource": true
          },
          "has_response": false,
          "is_composed": false
        }
      ]
    },
    {
      "name": "fidl.test.json/Child",
      "location": {
        "filename": "inheritance_with_recursive_decl.test.fidl",
        "line": 7,
        "column": 10,
        "length": 5
      },
      "methods": [
        {
          "ordinal": 2566928499103139852,
          "name": "First",
          "location": {
            "filename": "inheritance_with_recursive_decl.test.fidl",
            "line": 4,
            "column": 5,
            "length": 5
          },
          "has_request": true,
          "maybe_request": [
            {
              "type": {
                "kind": "request",
                "subtype": "fidl.test.json/Parent",
                "nullable": false
              },
              "name": "request",
              "location": {
                "filename": "inheritance_with_recursive_decl.test.fidl",
                "line": 4,
                "column": 27,
                "length": 7
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 4
              }
            }
          ],
          "maybe_request_payload": "fidl.test.json/SomeLongAnonymousPrefix0",
          "maybe_request_type_shape_v1": {
            "inline_size": 24,
            "alignment": 8,
            "depth": 0,
            "max_handles": 1,
            "max_out_of_line": 0,
            "has_padding": true,
            "has_flexible_envelope": false,
            "is_resource": true
          },
          "has_response": false,
          "is_composed": true
        },
        {
          "ordinal": 2628306304378964143,
          "name": "Second",
          "location": {
            "filename": "inheritance_with_recursive_decl.test.fidl",
            "line": 9,
            "column": 5,
            "length": 6
          },
          "has_request": true,
          "maybe_request": [
            {
              "type": {
                "kind": "request",
                "subtype": "fidl.test.json/Parent",
                "nullable": false
              },
              "name": "request",
              "location": {
                "filename": "inheritance_with_recursive_decl.test.fidl",
                "line": 9,
                "column": 28,
                "length": 7
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 4
              }
            }
          ],
          "maybe_request_payload": "fidl.test.json/SomeLongAnonymousPrefix1",
          "maybe_request_type_shape_v1": {
            "inline_size": 24,
            "alignment": 8,
            "depth": 0,
            "max_handles": 1,
            "max_out_of_line": 0,
            "has_padding": true,
            "has_flexible_envelope": false,
            "is_resource": true
          },
          "has_response": false,
          "is_composed": false
        }
      ]
    }
  ],
  "service_declarations": [],
  "struct_declarations": [
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix0",
      "location": {
        "filename": "inheritance_with_recursive_decl.test.fidl",
        "line": 4,
        "column": 10,
        "length": 25
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "request",
            "subtype": "fidl.test.json/Parent",
            "nullable": false
          },
          "name": "request",
          "location": {
            "filename": "inheritance_with_recursive_decl.test.fidl",
            "line": 4,
            "column": 27,
            "length": 7
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 4
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 8,
        "alignment": 8,
        "depth": 0,
        "max_handles": 1,
        "max_out_of_line": 0,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": true
      }
    },
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix1",
      "location": {
        "filename": "inheritance_with_recursive_decl.test.fidl",
        "line": 9,
        "column": 11,
        "length": 25
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "request",
            "subtype": "fidl.test.json/Parent",
            "nullable": false
          },
          "name": "request",
          "location": {
            "filename": "inheritance_with_recursive_decl.test.fidl",
            "line": 9,
            "column": 28,
            "length": 7
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 4
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 8,
        "alignment": 8,
        "depth": 0,
        "max_handles": 1,
        "max_out_of_line": 0,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": true
      }
    },
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix0",
      "location": {
        "filename": "inheritance_with_recursive_decl.test.fidl",
        "line": 4,
        "column": 10,
        "length": 25
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "request",
            "subtype": "fidl.test.json/Parent",
            "nullable": false
          },
          "name": "request",
          "location": {
            "filename": "inheritance_with_recursive_decl.test.fidl",
            "line": 4,
            "column": 27,
            "length": 7
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 4
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 8,
        "alignment": 8,
        "depth": 0,
        "max_handles": 1,
        "max_out_of_line": 0,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": true
      }
    }
  ],
  "table_declarations": [],
  "union_declarations": [],
  "type_alias_declarations": [],
  "declaration_order": [
    "fidl.test.json/Parent",
    "fidl.test.json/Child"
  ],
  "declarations": {
    "fidl.test.json/Parent": "interface",
    "fidl.test.json/Child": "interface",
    "fidl.test.json/SomeLongAnonymousPrefix0": "struct",
    "fidl.test.json/SomeLongAnonymousPrefix1": "struct"
  }
}
